File: src/bp20/refapp/trunk/microsoft/wcf/rivta-bp20-refapp/rivta-bp20-refapp-producer/Responder.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using rivta_bp20_refapp_schemas.wsdl;

namespace rivta_bp20_refapp_producer
{
    class Responder : EhrExtractionResponderInterface
    {
        public GetEhrExtractResponse GetEhrExtract(GetEhrExtractRequest request)
        {
            GetEhrExtractResponse response = new GetEhrExtractResponse();
            EHR_EXTRACT[] list = new EHR_EXTRACT[1];
            response.ehr_extract = list;

            EHR_EXTRACT e = createEhrExtract(request);

            list[0] = e;
            return response;
        }

        private EHR_EXTRACT createEhrExtract(GetEhrExtractRequest request)
        {
            EHR_EXTRACT e = new EHR_EXTRACT();

            II ap = new II();
            ap.root = "AuthorisingParty-UID";
            e.authorising_party = ap;

            II eid = new II();
            eid.root = "EhrId-UID";
            e.ehr_id = eid;

            II es = new II();
            es.root = "EhrSystem-UID";
            e.ehr_system = es;

            e.rm_id = "RmId";

            // relay something back that can be verified at the original requester (end user)
            II soc = new II();
            soc.root = "SubjectOfCare-UID";
            soc.flavorId = "sugar";
            II socRequest = request.GetEhrExtract.subject_of_care_id;
            String identifierName = socRequest.root;
            soc.identifierName = identifierName;
            e.subject_of_care = soc;

            TS tc = new TS();
            tc.value = DateTime.Now.ToString();
            e.time_created = tc;
            return e;
        }

        public GetEhrExtractContinuationResponse GetEhrExtractContinuation(GetEhrExtractContinuationRequest request)
        {
            throw new NotImplementedException();
        }

        public PingResponse Ping(PingRequest request)
        {
            PingResponse response = new PingResponse();
            response.info = request.Ping.info;
            response.logicalAddress = request.To.Value;
            return response;
        }
    }
}

File: RefApp/rivta-bp-21-ip/java/cxf/trunk/rivta-bpip21-refapp-producer/src/main/java/se/skl/rivta/bpip21/refapp/producer/makebooking/Responder.java
/**
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements. See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership. The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License. You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied. See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

package se.skl.rivta.bpip21.refapp.producer.makebooking;

import javax.jws.WebService;

import oasis.names.tc.saml._2_0.assertion.AssertionType;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import se.skl.crm.scheduling.v1.ResultCodeEnum;
import se.skl.riv.crm.scheduling.makebooking.v1.rivtabpip21.MakeBookingResponderInterface;
import se.skl.riv.crm.scheduling.makebookingresponder.v1.MakeBookingResponseType;
import se.skl.riv.crm.scheduling.makebookingresponder.v1.MakeBookingType;
import se.skl.rivta.bp20.refapp.util.Util;

@WebService(
	serviceName = "MakeBookingResponderService", 
	endpointInterface="se.skl.riv.crm.scheduling.makebooking.v1.rivtabpip21.MakeBookingResponderInterface", 
	portName = "MakeBookingResponderPort", 
	targetNamespace = "urn:riv:crm:scheduling:MakeBooking:1:rivtabpip21",
	wsdlLocation = "schemas/interactions/MakeBookingInteraction/MakeBookingInteraction_1.1_rivtabpip21.wsdl")
public class Responder implements MakeBookingResponderInterface {

	static private final Logger logger = LoggerFactory.getLogger(Util.class);

	@Override
	public MakeBookingResponseType makeBooking(String logicalAddress,
			AssertionType hcAssertion, MakeBookingType parameters) {
		
		logger.info("Tog emot anrop. Logisk adress {} Assertion: {}, Parameters: {}", new Object[]{logicalAddress, hcAssertion, parameters});
		
		
		final MakeBookingResponseType response = new MakeBookingResponseType();
		
		response.setBookingId("1");
		response.setResultCode(ResultCodeEnum.OK);
		response.setResultText("Bokning genomfšrd.");
		
		return response;
	}



//	public GetEhrExtractResponseType getEhrExtract(
//			AttributedURIType logicalAddress,
//			AssertionType hcAssertion,
//			GetEhrExtractRequestType parameters) {
//
//		logger.info("Operation getEhrExtract called");
//		try {
//			GetEhrExtractResponseType response = new GetEhrExtractResponseType();
//
//			EHREXTRACT e = createEhrExtract(parameters);
//
//			response.getEhrExtract().add(e);
//
//			return response;
//
//		} catch (RuntimeException e) {
//			System.out.println("Error occured: " + e);
//			throw e;
//		}
//	}
//
//	public GetEhrExtractContinuationResponseType getEhrExtractContinuation(
//			AttributedURIType logicalAddress,
//			AssertionType hcAssertion,
//			GetEhrExtractContinuationRequestType parameters) {
//
//		logger.info("Operation getEhrExtractContinuation called");
//
//		GetEhrExtractContinuationResponseType response = new GetEhrExtractContinuationResponseType();
//		return response;
//	}
//
//	public PingResponseType ping(AttributedURIType logicalAddress, PingRequestType parameters) {
//		logger.info("Operation ping called");
//		PingResponseType response = new PingResponseType();
//		response.setInfo(parameters.getInfo());
//		response.setLogicalAddress(logicalAddress.getValue());
//		return response;
//	}
//
//	private EHREXTRACT createEhrExtract(GetEhrExtractRequestType parameters) {
//		EHREXTRACT e = new EHREXTRACT();
//
//		II ap = new II();
//		ap.setRoot("AuthorisingParty-UID");
//		e.setAuthorisingParty(ap);
//
//		II eid = new II();
//		eid.setRoot("EhrId-UID");
//		e.setEhrId(eid);
//
//		II es = new II();
//		es.setRoot("EhrSystem-UID");
//		e.setEhrSystem(es);
//
//		e.setRmId("RmId");
//
//		// relay something back that can be verified at the original
//		// requester (end user)
//		II soc = new II();
//		soc.setRoot("SubjectOfCare-UID");
//		soc.setFlavorId("sugar");
//		II socRequest = parameters.getSubjectOfCareId();
//		String identifierName = socRequest.getRoot();
//		soc.setIdentifierName(identifierName);
//		e.setSubjectOfCare(soc);
//
//		TS tc = new TS();
//		tc.setValue(new Date().toString());
//		e.setTimeCreated(tc);
//		return e;
//	}

}